src/main/java/com/google/gwt/dom/client/DOMImpl.java

```diff
@@ -19,15 +19,108 @@
 
 import com.google.common.base.MoreObjects;
 import com.google.gwt.core.client.JsArray;
+import com.google.gwt.event.dom.client.BlurEvent;
+import com.google.gwt.event.dom.client.ChangeEvent;
 import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.FocusEvent;
+import com.google.gwt.event.dom.client.DomEvent.Type;
+import com.google.gwt.user.client.DOM;
+import com.google.gwt.user.client.Event;
+import com.google.gwt.user.client.EventListener;
 
 public class DOMImpl {
 	
 	static final DOMImpl impl = new DOMImpl();
 	
-	public void buttonClick(ButtonElement button) {
-		button.fireEvent(ClickEvent.getType());
-	}
+	// Element In Focus (some browsers actually have working implementation document.activeElement)
+	private Element activeElement;
+	
+    private void switchFocusTo(Element element) {
+        if (activeElement == element) {
+            return;
+        } else {
+            if (activeElement != null && isAttached(activeElement)) {
+                if (activeElement instanceof InputElement) {
+                    fireValueChangeIfAny((InputElement)activeElement);
+                }
+                activeElement.fireEvent(BlurEvent.getType());
+            }
+            element.fireEvent(FocusEvent.getType());
+            activeElement = element;
+        }
+    }
+
+    private boolean isAttached(Element elem) {
+        return DOM.getEventListener(elem) != null;
+    }
+
+    private void fireValueChangeIfAny(InputElement element) {
```

Comment:
Exactly this is what I want to avoid!  I want users of the library to be able to start using it right away without creating multiple helper classes. Now I also have large Helper class in my project.  I don't believe it is good approach to force users to create  helper classes.   On another hand they should be able to extend the helper classes but this is different story.

So I see the current implementation .  I see you have DOMImpl.buttonClick.
How is my additions are different ?